core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -369,4 +371,14 @@ default ClientBuilderParams validateParams(ClientBuilderParams params) {
         }
         return params;
     }
+
+    /**
+     * Add a shutdown hook to stop this {@link ClientFactory}.
+     */
+    default void closeOnShutdown() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
How about naming the thread? e.g. `armeria-shutdown-hook-NNN`

Reply:
It seems good.

Reply:
Do we decide not to name the thread? 🤔

Reply:
We should use it 🙇‍♂️